secretmgmt/secrets.go

```diff
@@ -41,15 +41,11 @@ func NewSecretResolver(resolveFunc ResolveSecretFunc, resolveTimeout time.Durati
 	return &SecretResolver{Resolve: resolveFunc, resolveTimeout: resolveTimeout}, nil
 }
 
-// ResolveSecrets returns a list of resolved secrets and returns error if there is any failure in resolving a secret.
-func (secretResolver *SecretResolver) ResolveSecrets(ctx context.Context, secrets []*graph.Secret) (Values, error) {
-
-	resolvedSecrets := Values{}
-
+// ResolveSecrets returns a map of resolved secrets (secretID:secretValue), and returns error if there is any failure in resolving a secret.
```

Comment:
could you explain why you wanted to shift to using a bool instead of writing the value and not return the list of values and instead set it higher up?
also this comment is out of date

Reply:
updated the comment.
As for using bool, we just wanted a signal to say the goroutine has been completed. We set the resolved value in the goroutine itself...